File: BOTAIML.VisionBot.Monitoring.Web/ViewModels/SubCenterAreaViewModel.cs
using BOTAIML.VisionBot.Monitoring.Web.Models;
using System;
using System.ComponentModel.DataAnnotations;

namespace BOTAIML.VisionBot.Monitoring.Web.ViewModels
{
    public class SubCenterAreaViewModel
    {
        [Required]
        public AreaName Name { get; set; }
        public DoorStatus? DoorStatus { get; set; }
        public int? PersonId { get; set; }
        public DateTime Time { get; set; }
        public float? Probability { get; set; }
        public PowerStatus? PowerStatus { get; set; }
        public TrackingStatus? TrackingStatus { get; set; }
        [Required]
        public int CameraId { get; set; }
        [Required]
        public bool IsAlertRequired { get; set; }
        public AlertContentType ContentType { get; set; }
        public AlertType Type { get; set; }
        public AlertLevel Level { get; set; }
        public string Message { get; set; }
        public string MediaFilePath { get; set; }
        public string Base64Data { get; set; }
    }
}

File: BOTAIML.VisionBot.Monitoring.Web/ClientApp/src/app/api/models/sub-center-area-view-model.ts
/* tslint:disable */
/* eslint-disable */
import { AlertContentType } from './alert-content-type';
import { AlertLevel } from './alert-level';
import { AlertType } from './alert-type';
import { AreaName } from './area-name';
import { DoorStatus } from './door-status';
import { PowerStatus } from './power-status';
import { TrackingStatus } from './tracking-status';
export interface SubCenterAreaViewModel {
  cameraId: number;
  contentType?: AlertContentType;
  doorStatus?: DoorStatus;
  isAlertRequired: boolean;
  level?: AlertLevel;
  mediaFilePath?: null | string;
  message?: null | string;
  name: AreaName;
  personId?: null | number;
  powerStatus?: PowerStatus;
  probability?: null | number;
  time?: string;
  trackingStatus?: TrackingStatus;
  type?: AlertType;
}
